src/tls/openssl.c

```diff
@@ -509,6 +509,7 @@ static int tls_net_handshake(struct flb_tls *tls,
 {
     int ret = 0;
     char err_buf[256];
+    char * target_host;
```

Comment:
I think we have a bug in https://github.com/fluent/fluent-bit/blob/f16d1c02fe92fb27b047a3da01e2f4b1d13d8ba6/src/tls/flb_tls.c#L503 because if `tls->vhost` is not `NULL`  then `vhost` is left as `NULL` which doesn't seem like the correct idea which in turn forces you to do it here.

What do you think of fixing that statement so it looks something like : 

```
if (connection->upstream->proxied_host != NULL) {
    vhost = connection->upstream->proxied_host;
}
else if (tls->vhost == NULL) {
    vhost = connection->upstream->tcp_host;
}
else {
    vhost = tls->vhost;
}
```

Then we can get rid of that.